torch/_library/utils.py

```diff
@@ -201,6 +201,44 @@ def zip_schema(
     return
 
 
+def hop_schema_from_fx_node(node):
+    from torchgen.gen_schema_utils import FunctionSchemaGen
+
+    hop = node.target
+    if not isinstance(hop, torch._ops.HigherOrderOperator):
+        raise RuntimeError("fx_node's target must be a hop.")
```

Comment:
Does this work if the HOP takes a default value as input?

Also, if a HOP takes a default value as input, we shouldn't trace it into the graph (as per the rules for aten operators). I don't know if that's what happens today, we should file a follow-up issue for this if not

Reply:
> if a HOP takes a default value as input, we shouldn't trace it into the graph

Each hop has its own logic nowadays. FlexAttention has a default kwarg `score_mod_other_buffers = tuple()` and it's currently traced into the graph. AssociativeScan is going to have a default kwarg reverse=False.

>  If someone adds a new default kwarg for a HOP

I'll write down my understanding about BC/FC: 

Yeah, this should work today. I can try to add a test for this but this should be after we actually use the schema gen mechanism for the control flow operators in export. 

**For BC:**
For example, associtive_scan adds `reverse=False`.The newer export receives an old hop without the default kwarg in its schema:
```
associative_scan(GraphModule combine_fn, Tensor[] input, int dim) -> Tensor[]
```
The new runtime is able to deserialize it by mapping the kwargs combined_fn, input and dim with proper inputs. For anything missing, (e.g. reverse=False), it will leave as it is. So we have BC in this case.
Case 2:  changing arg name name breaks BC:
For example, associative_scan changes `combine_fn` to `combine_graph` in its signature. The deserializer won't be able to map the `combine_fn` in old artifact to `combine_graph`. So this is BC-breaking.

**For FC:**
I think export generally doesn't care a lot about FC (at least that's my impression). Supposing a new artifact has `reverse=True` argument in associative_scan, when an old runtime tries to deserialize it, the `reverse=True` won't work if we create an associative_scan without the reverse argument. This means FC is broken if `reverse` is not the default value.

If `reverse=False`, (i.e. the BC-preserving change), the FC-story would be, we shouldn't add `reverse=False` into the graph. Then an older runtime can deserialize the op using the old one without any issue.

So concretely, i think we should:
- Do the`if a HOP takes a default value as input, we shouldn't trace it into the graph`
- Add a BC test when serde starts to use the generated schema.

Reply:
> So concretely, i think we should:

Yeah, that's how our current FC/BC things work for regular operators